scripts/prepare_sector_network.py

```diff
@@ -3391,6 +3396,73 @@ def add_offshore_hubs_tyndp(
     add_offshore_grid_tyndp(n, pyear, offshore_grid_fn, costs, nyears)
 
 
+def attach_gas_load(
+    n: pypsa.Network,
+    gas_demand_fn: str,
+    options: dict,
+    costs: pd.DataFrame,
+    spatial: SimpleNamespace,
+    nhours: int = 8760,
+):
+    """
+    Attach exogenous gas demand to the network as Load with CO2 emissions tracking.
+
+    Creates 'gas for demand' buses, loads, and links that consume gas from
+    main gas buses and emit CO2 to atmosphere based on gas carbon intensity.
+
+    Parameters
+    ----------
+    n : pypsa.Network
+        The network object to add offshore hubs and grid to.
+    gas_demand_fn : str
+        Exogenous gas demand file name.
+    options : dict
+        Dictionary of configuration options including:
+        - gas_demand_exogenously
+    costs : pd.DataFrame
+        Technology costs assumptions.
+    spatial : object, optional
+        Object containing spatial information about nodes and their locations.
+    nhours : int
+        Number of hours over which the annual gas demand is divided.
+    """
+
+    gas_demand = pd.read_csv(gas_demand_fn, index_col=0) / nhours
+
+    if options["gas_network"]:
+        spatial_gas_demand = gas_demand.rename(index=lambda x: x + " gas for demand")
+    else:
+        spatial_gas_demand = gas_demand.sum().rename({"p_nom": "EU gas for demand"})
+
+    n.add(
+        "Bus",
+        spatial.gas.exo_demand,
+        location=spatial.gas.locations,
+        carrier="gas for demand",
+        unit="MWh_LHV",
+    )
+
+    n.add(
+        "Load",
+        spatial.gas.exo_demand,
+        bus=spatial.gas.exo_demand,
+        carrier="gas for demand",
+        p_set=spatial_gas_demand,
+    )
+
+    n.add(
+        "Link",
+        spatial.gas.exo_demand,
+        bus0=spatial.gas.nodes,
+        bus1=spatial.gas.exo_demand,
+        bus2="co2 atmosphere",
+        carrier="gas for demand",
+        p_nom_extendable=True,
```

Comment:
Since the link does not have any costs, we can set
```suggestion
        p_nom=np.inf,
```